sdks/java/container/Dockerfile-java11

```diff
@@ -16,6 +16,10 @@
 # limitations under the License.
 ###############################################################################
 
+##
+# NOTE: This image is now deprecated.
+# Use Dockerfile with the appropriate java_version build argument.
+##
```

Comment:
I'm +1 for just removing this. Looks like it was added in https://github.com/apache/beam/pull/8053 in order to start running a job for testing Java 11. As long as the jobs that use Java 11 still pass after removing this I think it's fine.

I don't think users should be referencing this Dockerfile directly, if they're building a Java 11 container at all it should be through the gradle command (and even then we don't support Java 11 so they shouldn't be doing it).

Reply:
Also the fact that this Dockerfile doesn't copy LICENSE and NOTICE is problematic.